python_modules/dagster/dagster/_core/ext/utils.py

```diff
@@ -163,3 +190,24 @@ def extract_message_or_forward_to_stdout(ext_context: "ExtOrchestrationContext",
     except Exception:
         # move non-message logs in to stdout for compute log capture
         sys.stdout.writelines((log_line, "\n"))
+
+
+@contextmanager
+def setup_ext_protocol(
+    context: OpExecutionContext,
+    context_injector: ExtContextInjector,
+    message_reader: ExtMessageReader,
+    extras: Optional[ExtExtras] = None,
+) -> Iterator[Mapping[str, str]]:
+    """Enter the context managed context injector and message reader that power the EXT protocol and receive the environment variables
+    that need to be provided to the external process.
+    """
+    ext_context = ExtOrchestrationContext(context=context, extras=extras)
+    with context_injector.inject_context(ext_context) as ci_params, message_reader.read_messages(
+        ext_context,
+    ) as mr_params:
+        protocol_envs = {
+            DAGSTER_EXT_ENV_KEYS["is_orchestration_active"]: encode_env_var(True),
+            **io_params_as_env_vars(ci_params, mr_params),
```

Comment:
I think we should yield these params in their "unencoded form" (i.e. not env vars yet) and leave to the launcher to encode as env vars.

Reply:
Given all the call sites we have so far I think theres a nice simplicity to having a fn that just gives you the one bundle of env vars to pass along. 

I would be fine decomposing this one function in to a base one that yields params and a separate one that calls the base and then bundles all the env vars that most usage would go through. Not sure the right naming scheme though 

Does the databricks one use the raw params?

Reply:
Alternatively we could have this return an object and offer a couple methods. I bet we will end up needing more stuff here anyway, so wrapping it all up in an object could be nice